lua_api.txt: more detailed documentation of 'set_attach' method. (#11632)
minetest/minetest: doc/lua_api.txt

object you are working with still exists.

> ### Attachment
> 
> It is possible to attach objects to other objects (`set_attach` method).
> 
> When an object is attached, its position gets relative to the parent`s one and
> rotation is a sum of its own, the parent`s and relative rotations. Standard
> methods like 'set_pos' and 'set_rotation' won`t work, you should call repeatedly
> `set_attach` with already new position or rotation tables.
> 
> **Bug**: keep in mind to multiply relative position by 10 when attaching to make it actually proper!
> 
> Example:
> `child:set_attach(parent, "", vector.multiply(vector.subtract(child:get_pos(), parent:get_pos()), 10))`.
> 
> It is also possible to set bones for parent objects like in Blender (`set_bone_position` method)
> and use them for child ones to adjust their position/rotation. After assignment a bone to an attached object,
> it will follow movement and rotation of that bone.

Reviewer: ```suggestion
### Attachments

It is possible to attach objects to other objects (`set_attach` method).

When an object is attached, it is positioned relative to the parent's position and rotation.
`get_pos` and `get_rotation` will always return the parent's values and can't be set directly.
To change position or rotation call `set_attach` again with the new values.

**Note**: Just like model dimensions, the relative position in `set_attach` must be multiplied by 10.

It is also possible to attach to a bone of the parent object (as defined in the model).
After assignment a bone to an attached object, the child will follow movement and rotation of that bone.
```

Author: I think it is necessary to mention also the child rotation is a sum of its own, the parent's snd relative rotations, at least I checked it experimentally.

* `set_animation_frame_speed(frame_speed)`
    * `frame_speed`: number, default: `15.0`
* `set_attach(parent[, bone, position, rotation, forced_visible])`
	* `bone`: string.

> 		* This is a name of the parent's bone that is set with `set_bone_position`.

Reviewer 2: isn't it from the model?

		* After setting, the child will follow the movement and rotation of that bone.
		* Default is "", the root bone.
	* `position`: {x=num, y=num, z=num}.
		* Relative to the parent's model space.
		* Default is {x=0, y=0, z=0}.
		* Note: Just like model dimensions, it should be multiplied by 10

> 		  to set it actually proper.

Reviewer 2: ```suggestion
		 compared to a world position.
```

	* `rotation`: {x=num, y=num, z=num}.

> 		* Rotation around each axis in degrees.

Reviewer 2: ```suggestion
		* Rotation around each axis in degrees, relative to parent's rotation.
```
